controllers/cloud.redhat.com/providers/kafka/provider.go

```diff
@@ -14,7 +14,7 @@ import (
 // KafkaProvider is the interface for apps to use to configure kafka topics
 type KafkaProvider interface {
 	p.Configurable
-	CreateTopic(nn types.NamespacedName, topic *strimzi.KafkaTopicSpec) error
+	CreateTopic(nn types.NamespacedName, topic *strimzi.KafkaTopicSpec, app *crd.ClowdApp) error
```

Comment:
Doesn't `nn` represent the `NamespacedName` of the app that you're now passing in?

Reply:
If you're going to pass the app in, I suggest you remove the other two arguments and change the name to `CreateTopics` and then iterate over the topics in each kafka mode.